### ISSUE TITLE
Incorrect EE definition detection for collection meta/execution-environment.yml

### ISSUE THREAD

**Issue by dbrennand** [ansible/ansible-lint#4710]
##### Summary

In an Ansible collection, Ansible Lint is detecting the `meta/execution-environment.yml` file as an Execution Environment definition when it is actually collection level metadata. 

Please see below for context:

- https://github.com/ansible-collections/collection_template/pull/89#discussion_r2250380750 
- https://github.com/ansible-collections/collection_template/pull/89

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
❯ uv run ansible-lint --version    
ansible-lint 25.7.0 using ansible-core:2.19.0 ansible-compat:25.6.0 ruamel-yaml:0.18.14 ruamel-yaml-clib:0.2.12
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

<img width="817" height="377" alt="Image" src="https://github.com/user-attachments/assets/6d3793d8-719e-461d-bbd8-f5730628920e" />

##### Desired Behavior

Ansible Lint does not detect this file as an Execution Environment Definition and does not raise the below error.

##### Actual Behavior

```paste below
schema[execution-environment]: $ 'version' is a required property. See https://ansible-builder.readthedocs.io/en/latest/definition/
meta/execution-environment.yml:1
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve
